### PR TITLE
ET-3359 API gateway setup

### PR DESCRIPTION
**Pull request by Robert-Steiner**
## What 🕵️ 🔍

- api gateway setup with authorizer and usage plan.
- uses the current openapi spec to define the resources and methods

## References 📝 🔗

- [ET-3359]

## Reviewer checklist

*Reviewer agreement:*

* Reviewers shouldn't review a "Draft Pull Request", but rather wait for it to be marked as "Ready for Review".
* Reviewers do **not** commit or merge the pull request.
* If/when there are changes that need to be applied (applying Terraform or deploying Kubernetes manifests), the **assignee** is the one responsible for doing it.
* Use the labels `not-applied`, `partially-applied` and `fully-applied` so others know the current progress.
  * Remember to change labels accordingly once you start and finish to apply the changes.

### CODE REVIEW

**Comment by Robert-Steiner** [comment]
I moved the usage plan code in a separate branch because i don't think it is worth to merge it but we can show it in the demo.

File: aws/stage/lambda/main.tf [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  }
+}
+
+provider "aws" {
+  region = "eu-central-1"
+}
+
+locals {
+  authorizer_function_name = "authorizer"
+  authorizer_output_path   = "${path.module}/authorizer/lambda.zip"
```
Comment (jorge-cr-13) [inline aws/stage/lambda/main.tf:27]: Where are we storing these `.zip` files and what do they contain?

File: aws/stage/lambda/variables.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+variable "api_key" {
```
Comment (jorge-cr-13) [inline aws/stage/lambda/variables.tf:1]: How are we generating this key?

File: aws/stage/lambda/main.tf [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  hello_world_function_name = "hello_world"
+  hello_world_output_path   = "${path.module}/hello_world/lambda.zip"
+}
+
+# `authorizer` lambda
+resource "local_sensitive_file" "authorizer_code" {
+  content = templatefile("${path.module}/authorizer/index.js.tpl", {
+    api_key : var.api_key,
+  })
+  filename = "${path.module}/authorizer/index.js"
```
Comment (jorge-cr-13) [inline aws/stage/lambda/main.tf:38]: What is this file?

Comment (Robert-Steiner) [reply aws/stage/lambda/main.tf:38]: The file contains the code for the authorizer lambda. what we do here is, we take the `aws/stage/lambda/authorizer/index.js.tpl` template + the api key and generate the `index.js` file. this way we don't hardcode the api key into the file.
the `index.js` file is then used in the code blow:
1. take the `index.js` file and zip it. 
2. upload the zip to aws lambda

File: aws/stage/lambda/main.tf (hunk 1 above)
Comment (Robert-Steiner) [reply aws/stage/lambda/main.tf:27]: we store them locally and they contain the code for the authorizer and hello world lambda
the zip files will be uploaded to aws lambda

File: aws/stage/lambda/variables.tf (hunk 2 above)
Comment (Robert-Steiner) [reply aws/stage/lambda/variables.tf]: We can have aws generate the key for us, or we can do this by generating a string via 1password, for example. tbh, i though we need to have an api gateway first before we can generate a key but that is not true. i removed the variable and let aws generate the key for us.

**Review by jorge-cr-13 — APPROVED** [review]
LGTM! 👍🏻

**Pull request merged**